cv19ResSupportV3.Tests/V3/Gateways/ResidentGatewayTest.cs

```diff
@@ -5,89 +5,477 @@
-            var createdRecord = DatabaseContext.ResidentEntities.Find(response);
-            var expectedRecord = new ResidentEntity
+
+            // act
+            var duplicateResidentIdAttempt1 = _classUnderTest.FindResident(searchParametersNoFirstName);
+            var duplicateResidentIdAttempt2 = _classUnderTest.FindResident(searchParametersNoLastName);
+            bool isResidentRequest1Duplicate = duplicateResidentIdAttempt1 != null;
+            bool isResidentRequest2Duplicate = duplicateResidentIdAttempt2 != null;
+
+            // assert
+            isResidentRequest1Duplicate.Should().BeFalse();
+            duplicateResidentIdAttempt1.Should().Be(null);
+
+            isResidentRequest2Duplicate.Should().BeFalse();
+            duplicateResidentIdAttempt2.Should().Be(null);
+        }
+
+        [Test]
+        public void FindResidentReturnsAMatchByNhsNumberEvenWhenProvidedNhsNumberIsNotTrimmed()
+        {
+            // arrange
+
+            // Case 1 setup: Non-trimmed number is in the request, and the trimmed one in the DB.
+
+            var nhsNumber1 = _faker.Random.Hash();
+            var matchingNonTrimmedNHSNumber1 = $"  {nhsNumber1}     ";
+
+            // don't care about Fname, Lname or anything else, as I'm expecting a match via NHS number
+            var searchParameters1 = new FindResident
             {
-                Id = existingResident.Id,
-                FirstName = "FirstName",
-                LastName = "LastName",
-                DobDay = "Day",
-                DobMonth = "Month",
-                DobYear = "Year"
+                NhsNumber = matchingNonTrimmedNHSNumber1
             };
 
-            createdRecord.Should().BeEquivalentTo(expectedRecord);
+            var existingResidentWithNormalNhsNumber = new ResidentEntity
+            {
+                NhsNumber = nhsNumber1
+            };
+
+            // Case 2 setup: Non-trimmed number is in the DB, and the trimmed one in the request.
+
+            var matchingNhsNumber2 = _faker.Random.Hash();
+            var nonTrimmedNHSNumber2 = $"     {matchingNhsNumber2} ";
+
+            var searchParameters2 = new FindResident
+            {
+                NhsNumber = matchingNhsNumber2
+            };
+
+            var existingResidentWithNonTrimmedNhsNumber = new ResidentEntity
+            {
+                NhsNumber = nonTrimmedNHSNumber2
+            };
+
+            DatabaseContext.ResidentEntities.Add(existingResidentWithNormalNhsNumber);
+            DatabaseContext.ResidentEntities.Add(existingResidentWithNonTrimmedNhsNumber);
+            DatabaseContext.SaveChanges();
+
+            var ResidentWithNormalNhsNumberId1 = DatabaseContext.ResidentEntities
+                .FirstOrDefault(r => r.NhsNumber == nhsNumber1)?.Id;
+
+            var ResidentWithNormalNhsNumberId2 = DatabaseContext.ResidentEntities
+                .FirstOrDefault(r => r.NhsNumber == nonTrimmedNHSNumber2)?.Id;
+
```

Comment:
Can it be ```existingResidentWithNormalNhsNumber.Id``` and ```existingResidentWithNonTrimmedNhsNumber.Id``` ?

Reply:
Same answer as for the first 2 cases of this 😁 

We have 2 routes here:
1) Guess an Id and hardcode it.
2) Use ".FirstOrDefault()" Linq expression, which is already tested by Microsoft, and doesn't impact what's being tested.

I feel the later is more elegant and safer solution.

Reply:
Nvm... adding a record to DB has a side effect that updates the object that you're inserting. Good shout!